### PR TITLE
Adding ip validation for primary ip file and new unitests

### PR DESCRIPTION
**Pull request by tsorya**
Adding ip validation for primary ip file if it will be changed in future
Adding internal folder to unitests
Adding new unitests

### CODE REVIEW

**Comment by donpenney** [comment]
@tsorya it looks like your update triggers a change in the go.mod file, which isn't included in the commit. The ci-job test will run steps like "go mod tidy", and it looks like a dependency on github.com/pkg/errors went from indirect to direct

File: internal/clusterconfig/networkconfig.go [hunk 1]
```diff
@@ -29,9 +38,14 @@ func (r *UpgradeNetworkConfigGather) FetchNetworkConfig(ctx context.Context, ost
...
 
+	err = r.validateIPAddressFile()
+	if err != nil {
+		return err
+	}
+
 	for _, path := range listOfPaths {
 		r.Log.Info("Copying network files", "file", path, "to", dir)
-		err = cp.Copy(path, filepath.Join(dir, filepath.Base(path)))
+		err = cp.Copy(filepath.Join(hostPath, path), filepath.Join(dir, filepath.Base(path)))
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/clusterconfig/networkconfig.go:48]: Changing the path in next PR for script, let me open it too

[the remark this answers is not in the document]
Comment (tsorya) [reply internal/clusterconfig/networkconfig.go:48]: https://github.com/openshift-kni/lifecycle-agent/pull/34

**Pull request merged**